WIP [readobj][AArch64] Parse AArch64 build attributes

sivan-shani opened the pull request:
Add support for parsing AArch64 build attributes in llvm-readobj

sivan-shani committed:
[readobj][AArch64] Support Parsing AArch64 Build Attributes

Added support for parsing AArch64 build attributes in llvm-readobj.

ostannard commented on llvm/lib/Support/AArch64AttributeParser.cpp:
```diff
@@ -0,0 +1,152 @@
+    }
+    // All public subsections names must be known
+    if (VendorName.starts_with("aeabi")) {
+      if (!("aeabi_feature_and_bits" == VendorName ||
```
We should be able to print any "aeabi*" subsection, because the format is known, even if the tag names aren't.

sivan-shani committed:
[AArch64][Build Attributes] Improve testing (delete previous tests files)

This commit deletes previous tests files, following commits adding new test files.

sivan-shani committed:
[AArch64][Build Attributes] Standardize parsing error messages

Parsing error message should not print the symbol triggering error.
(Only point to it with the symbol '^')

sivan-shani committed:
[AArch64][Build Attributes] Print Tags as Numbers with Comments

Print Tags as numbers followed by comments instead of using string representations.
For example:
.aeabi_attribute 0, 1 @ Tag_Feature_BTI

instead of:
.aeabi_attribute Tag_Feature_BTI, 1

This reduces dependencies between compiler and assembler versions.

sivan-shani committed:
[AArch64][Build Attributes] Fix Parsing of Some Tags and Values

- Reject strings for unknown tags.
- Allow any value for Tag_PAuth_Platform and Tag_PAuth_Schema.

sivan-shani committed:
[AArch64][Build Attributes] Remove Assertion for Duplicate Values

Adding the same value twice is valid, so the assertion has been removed.

sivan-shani committed:
[AArch64][Build Attributes] Add test files

Add test files

sivan-shani commented:
Changes submitted here:
https://github.com/llvm/llvm-project/pull/128727
Closing this PR
